src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -84,38 +84,46 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
 }, {});
 
 const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelect }) => {
-  const [isNewSelectedPath, setIsNewSelectedPath] = useState(false);
-  const [previousSelectedPath, setPreviousSelectedPath] = useState(selectedPath);
   const [openKeys, setOpenKeys] = useState(openKeysToItem(menuItems[0], selectedPath));
+  const [currentNodeId, setCurrentNodeId] = useState();
+  const cursor = useRef(0);
+  const visibleNodes = useRef();
   const selectedItem = useRef();
   const theme = useContext(ThemeContext);
-
-  if (previousSelectedPath !== selectedPath) {
-    setIsNewSelectedPath(true);
-    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
-    setPreviousSelectedPath(selectedPath);
-  }
+  visibleNodes.current = [];
 
   useEffect(() => {
     if (selectedItem && selectedItem.current) {
       selectedItem.current.scrollIntoView();
     }
   }, []);
 
-  useLayoutEffect(() => {
-    if (!isNewSelectedPath) {
-      return;
-    }
-    const currentItemPosition = selectedItem?.current ? selectedItem.current.getBoundingClientRect() : null;
+  useEffect(() => {
+    cursor.current = visibleNodes.current.findIndex((el) => el.id === currentNodeId);
+    const currentNode = currentNodeId ? document.querySelector(`#${currentNodeId}`) : null;
+    const currentNodePosition = currentNode ? currentNode.getBoundingClientRect() : null;
     const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
-    if (!currentItemPosition || !navigationMenuPosition) {
-      return;
+
+    if (currentNode) currentNode.focus();
+    // If the item selected through keyboard navigation is not visible, scroll it into view.
+    if (currentNode && currentNodePosition && (currentNodePosition.bottom > navigationMenuPosition.bottom || currentNodePosition.top < navigationMenuPosition.top)) {
+      currentNode.scrollIntoView();
     }
+  }, [currentNodeId]);
+
+  useEffect(() => {
+    const selectedItemPosition = selectedItem?.current ? selectedItem.current.getBoundingClientRect() : null;
+    const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
     // If the current item is not visible, scroll the item into view.
-    if (currentItemPosition.bottom > navigationMenuPosition.bottom || currentItemPosition.top < navigationMenuPosition.top) {
+    if (selectedItemPosition && navigationMenuPosition && (selectedItemPosition.bottom > navigationMenuPosition.bottom || selectedItemPosition.top < navigationMenuPosition.top)) {
       selectedItem.current.scrollIntoView();
     }
-  });
+  }, [openKeys]);
+
+  useEffect(() => {
+    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
```

Comment:
openKeysToItem is expecting a list as the first parameter, this be ?
```suggestion
    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems, selectedPath) });
```

Reply:
Collapsing Navigation Menu has expected its top level menu items to be the first element in the `menuItems` prop since I've started working with it, but I'm not exactly sure why. Until now I've just left it as is because changing it would require me to alter how its parent component passes it props.